iOS/LocalPackages/DataBrokerProtection-iOS/Sources/DataBrokerProtection-iOS/DataCaching/DataBrokerProtectionDataManager.swift

```diff
@@ -0,0 +1,442 @@
+//
+//  DataBrokerProtectionDataManager.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+import os.log
+import DataBrokerProtectionCore
+
+public extension Notification.Name {
+    /// Notification posted when a profile is saved.
+    static let pirProfileSaved = Notification.Name("pirProfileSaved")
+}
+
+/// A protocol that defines the behavior for posting a notification when a profile is saved, if permitted by certain conditions.
+public protocol DBPProfileSavedNotifier {
+
+    /// Posts the "Profile Saved" notification if certain conditions allow it.
+    /// This method checks whether the notification can be posted, and if so, it triggers the notification.
+    func postProfileSavedNotificationIfPermitted()
+}
+
+public protocol DataBrokerProtectionDataManaging {
+    var cache: InMemoryDataCache { get }
+    var delegate: DataBrokerProtectionDataManagerDelegate? { get set }
+
+    init(database: DataBrokerProtectionRepository,
+         profileSavedNotifier: DBPProfileSavedNotifier?)
+    func saveProfile(_ profile: DataBrokerProtectionProfile) async throws
+    func fetchProfile() throws -> DataBrokerProtectionProfile?
+    func prepareProfileCache() throws
+    func fetchBrokerProfileQueryData(ignoresCache: Bool) throws -> [BrokerProfileQueryData]
+    func prepareBrokerProfileQueryDataCache() throws
+    func hasMatches() throws -> Bool
+    /// Returns the total number of matches and brokers with matches.
+    func matchesFoundAndBrokersCount() throws -> (matchCount: Int, brokerCount: Int)
+    func profileQueriesCount() throws -> Int
+}
+
+public protocol DataBrokerProtectionDataManagerDelegate: AnyObject {
+    func dataBrokerProtectionDataManagerDidUpdateData()
+    func dataBrokerProtectionDataManagerDidDeleteData()
+    func dataBrokerProtectionDataManagerWillOpenSendFeedbackForm()
+    func dataBrokerProtectionDataManagerWillApplyVPNBypassSetting(_ bypass: Bool) async
+    func isAuthenticatedUser() -> Bool
+}
+
+public class DataBrokerProtectionDataManager: DataBrokerProtectionDataManaging {
```

Comment:
I of course have all sorts of thoughts about the data manager but for the purpose of what we're doing here, I think this is good enough.